filebeat/harvester/log.go

```diff
@@ -258,16 +248,20 @@ func (h *Harvester) close() {
 	// On completion, push offset so we can continue where we left off if we relaunch on the same file
 	h.sendStateUpdate()
 
-	logp.Debug("harvester", "Stopping harvester for file: %s", h.path)
+	logp.Debug("harvester", "Stopping harvester for file: %s", h.state.Source)
 
 	// Make sure file is closed as soon as harvester exits
 	// If file was never opened, it can't be closed
 	if h.file != nil {
 		h.file.Close()
-		logp.Debug("harvester", "Stopping harvester, closing file: %s", h.path)
+		logp.Debug("harvester", "Stopping harvester, closing file: %s", h.state.Source)
+		harvesterOpenFiles.Add(-1)
 	} else {
-		logp.Warn("Stopping harvester, NOT closing file as file info not available: %s", h.path)
+		logp.Warn("Stopping harvester, NOT closing file as file info not available: %s", h.state.Source)
 	}
+
+	harvesterClosed.Add(1)
+	harvesterRunning.Add(-1)
```

Comment:
counters at correct place (`harvesterRunning`)? Maybe add `harvesterRunning.Add(1); defer harvesterRunning.Add(-1)` to harvester run loop.